language/documentation/book/src/friends.md

```diff
@@ -0,0 +1,126 @@
+# Friends
+
+The `friend` syntax is used to declare modules that are trusted by the current module.
+A trusted module is allowed to call any function defined in the current module that have the `public(friend)` visibility.
+For details on function visibilities, please refer to the *Visibility* section in [Functions](./functions.md).
+
+## Friend declaration
+
+A module can declare other modules as friends via friend declaration statements, in the format of
+
+- `friend <address::name>` — friend declaration using fully qualified module name like the example below, or
+```move
+address 0x42 {
+module A {
+    friend 0x42::B;
+}
+}
+```
+- `friend <module-name-alias>` — friend declaration using a module name alias, where the module alias is introduced via the `use` statement.
+```move
+address 0x42 {
+module A {
+    use 0x42::B;
+    friend B;
+}
+}
+```
+
+A module may have multiple friend declarations, and the union of all the friend modules forms the friend list.
+In the example below, both `0x42::B` and `0x42::C` are considered as friends of `0x42::A`.
+```move
+address 0x42 {
+module A {
+    friend 0x42::B;
+    friend 0x42::C;
+}
+}
+```
+
+Unlike `use` statements, `friend` can only be declared in the module scope and not in the expression block scope.
+In theory, it is valid to have `friend` declarations at any location where a top-level construct (e.g., `use`, `function`, `struct`, etc) is allowed.
```

Comment:
There shouldn't be anything "theoretical" about it: either they're allowed or they're not. I think we can be more succinct here, too:

```suggestion
`friend` declarations may be located anywhere a top-level construct (e.g., `use`, `function`, `struct`, etc.) is allowed.
```